src/bp/ui-admin/src/Pages/Confusion/index.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { Component } from 'react'
+
+import SectionLayout from '../Layouts/Section'
+import Details from './details'
+import moment from 'moment'
+
+import _ from 'lodash'
+
+import api from '../../api'
+
+class Confusion extends Component {
+  state = {
+    confusions: [],
+    botIds: [],
+    select: [],
+    isComputing: false
+  }
+
+  getDateFromTimestamp = timestamp => moment(parseInt(timestamp, 10)).format('LLL')
+
+  setSelect = (value, index) =>
+    this.setState({ select: [...this.state.select.slice(0, index), value, ...this.state.select.slice(index + 1)] })
+
+  selectChangeFromFrontEnd = index => event => this.setSelect(event.target.value, index)
+
+  getAllBotIds = async () => {
+    if (_.isEmpty(this.state.botIds)) {
+      this.setState({ botIds: (await api.getSecured().get('/admin/bots')).data.payload.bots.map(bot => bot.id) })
+    }
+
+    return this.state.botIds
+  }
+
+  fetchConfusionHashForBot = async botId => (await api.getSecured().get(`/bots/${botId}/mod/nlu/confusion`)).data
+
+  triggerComputeConfusionForBot = async botId =>
+    (await api.getSecured({ timeout: 999999999 }).post(`/bots/${botId}/mod/nlu/confusion`)).data
```

Comment:
Try to wrap your api calls in a try catch, so you can handle errors. Please extract that timeout value in a constant, eg: NO_TIMEOUT = 99999